js/adapt-assessmentArticleModel.js

```diff
@@ -1,836 +1,826 @@
-          wereQuestionsRestored = false;
-          break;
-        }
-      }
+    const isPageReload = this._checkReloadPage();
 
-      if (!wereQuestionsRestored) {
-        this.set('_assessmentCompleteInSession', true);
-        return true;
+    // stop resetting if not complete or not allowed
+    if (this.get('_assessmentCompleteInSession') &&
+      !assessmentConfig._isResetOnRevisit &&
+      !isPageReload &&
+      !force) {
+      if (typeof callback === 'function') {
+        // eslint-disable-next-line node/no-callback-literal
+        callback(false);
       }
-
       return false;
-    },
-
-    // Public Functions
-    isAssessmentEnabled: function() {
-      if (this.get('_assessment') &&
-        this.get('_assessment')._isEnabled) return true;
-      return false;
-    },
-
-    canResetInPage: function() {
-      var assessmentConfig = this.getConfig();
-      if (assessmentConfig._reloadPageOnReset === false) return false;
-      return true;
-    },
-
-    reset: function(force, callback) {
-
-      if (this._isResetInProgress) {
-        // prevent multiple resets from executing.
-        // keep callbacks in queue for when current reset is finished
-        this.once('reset', function() {
-          this._isResetInProgress = false;
-          if (typeof callback === 'function') {
-            // eslint-disable-next-line standard/no-callback-literal
-            callback(true);
-          }
-        });
-        return;
-      }
-
-      var assessmentConfig = this.getConfig();
+    }
 
-      // check if forcing reset via page revisit or force parameter
-      force = this._forceResetOnRevisit || force === true;
-      this._forceResetOnRevisit = false;
+    // check if new session and questions not restored
+    const wereQuestionsRestored = this._checkIfQuestionsWereRestored();
+    force = force || wereQuestionsRestored;
+    // the assessment is going to be reset so we must reset attempts
+    // otherwise assessment may not be set up properly in next session
+    if (force && !this._isAttemptsLeft()) {
+      this.set({
+        _attemptsLeft: this.get('_attempts'),
+        _attemptsSpent: 0
+      });
+    }
 
-      var isPageReload = this._checkReloadPage();
+    const allowResetIfPassed = this.get('_assessment')._allowResetIfPassed;
+    // stop resetting if no attempts left and allowResetIfPassed is false
+    if (!this._isAttemptsLeft() && !force && !allowResetIfPassed) {
+      // eslint-disable-next-line node/no-callback-literal
+      if (typeof callback === 'function') callback(false);
+      return false;
+    }
 
-      // stop resetting if not complete or not allowed
-      if (this.get('_assessmentCompleteInSession') &&
-        !assessmentConfig._isResetOnRevisit &&
-        !isPageReload &&
-        !force) {
+    if (!isPageReload) {
+      // only perform this section when not attempting to reload the page
+      // wait for reset to trigger
+      this.once('reset', () => {
```

Comment:
Was it on purpose that an arrow function was introduced here but not on the subsequent lines, i.e. 657/661?